src/ringmesh/geomodel/core/geomodel_core_js.cpp

```diff
@@ -40,9 +40,21 @@
 
 namespace RINGMesh
 {
-    NBIND_CLASS( GeoModel3D )
+    NBIND_CLASS(GeoModelBase3D)
     {
+        method(name);
+        method(nb_mesh_entities);
+        method(nb_geological_entities);
+        method(nb_corners);
+        method(nb_lines);
+        method(nb_surfaces);
+    }
+
+    NBIND_CLASS(GeoModel3D)
+    {
+        inherit(GeoModelBase3D);
```

Comment:
OK for GM3D but you will have to do the same for GM2D after.

Reply:
Yes, not sure the best way to do it. I just want to have something working for 3D asap. We will think about 2D later.